tests/e2e/e2e_util_test.go

```diff
@@ -978,3 +978,78 @@ func (s *IntegrationTestSuite) executeRedelegate(c *chain, valIdx int, endpoint
 	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
 	s.T().Logf("%s successfully redelegated %s from %s to %s", delegatorAddr, amount, originalValOperAddress, newValOperAddress)
 }
+
+func (s *IntegrationTestSuite) execSetWithrawAddress(c *chain, valIdx int, endpoint, fees string, delegatorAddress string, newWithdrawalAddress string, homePath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("Setting distribution withdrawal address on chain %s for %s to %s", c.id, delegatorAddress, newWithdrawalAddress)
+	gaiaCommand := []string{
+		"gaiad",
+		"tx",
+		"distribution",
+		"set-withdraw-addr",
+		newWithdrawalAddress,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, delegatorAddress),
+		fmt.Sprintf("--%s=%s", flags.FlagGasPrices, fees),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, c.id),
+		fmt.Sprintf("--%s=%s", flags.FlagHome, homePath),
+		"--keyring-backend=test",
+		"--output=json",
+		"-y",
+	}
+
+	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
+	s.T().Logf("Successfully set new distribution withdrawal address for %s to %s", delegatorAddress, newWithdrawalAddress)
+}
+
+func (s *IntegrationTestSuite) execWithdrawReward(c *chain, valIdx int, endpoint, fees string, delegatorAddress string, validatorAddress string, homePath string) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
+	defer cancel()
+
+	s.T().Logf("Withdrawing distribution rewards on chain %s for delegator %s from %s validator", c.id, delegatorAddress, validatorAddress)
+	gaiaCommand := []string{
+		"gaiad",
+		"tx",
+		"distribution",
+		"withdraw-rewards",
+		validatorAddress,
+		fmt.Sprintf("--%s=%s", flags.FlagFrom, delegatorAddress),
+		fmt.Sprintf("--%s=%s", flags.FlagGasPrices, "300uatom"),
+		fmt.Sprintf("--%s=%s", flags.FlagGas, "auto"),
+		fmt.Sprintf("--%s=%s", flags.FlagGasAdjustment, "1.5"),
+		fmt.Sprintf("--%s=%s", flags.FlagChainID, c.id),
+		fmt.Sprintf("--%s=%s", flags.FlagHome, homePath),
+		"--keyring-backend=test",
+		"--output=json",
+		"-y",
+	}
+
+	s.executeGaiaTxCommand(ctx, c, gaiaCommand, valIdx, endpoint)
+	s.T().Logf("Successfully withdrew distribution rewards for delegator %s from validator %s", delegatorAddress, validatorAddress)
+}
+
+func (s *IntegrationTestSuite) queryGaiaTx(endpoint, txHash string) error {
+	resp, err := http.Get(fmt.Sprintf("%s/cosmos/tx/v1beta1/txs/%s", endpoint, txHash))
+	if err != nil {
+		return fmt.Errorf("failed to execute HTTP request: %w", err)
+	}
+
+	defer resp.Body.Close()
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("tx query returned non-200 status: %d", resp.StatusCode)
+	}
+
+	var result map[string]interface{}
```

Comment:
Unmarshalling all values on a map cannot be performative. Can we create a struct to wrap only the necessary values?

Reply:
See below comment